src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/scaleset/AzureScaleSetProvider.java

```diff
@@ -0,0 +1,327 @@
+ *
+ * This class has been tested by ActiveEon to be thread safe (using Azure SDK release version 1.0.0-beta5).
+ * However this need to be carefully double-checked after every SDK upgrades, as mentioned by Microsoft:
+ * ------------------------------------------------------------------------------------------------------------
+ * We do not make any thread-safety guarantees about our libraries. We also do not test them for thread-safety.
+ * Methods that are currently thread-safe may be thread-unsafe in future versions.
+ * ------------------------------------------------------------------------------------------------------------
+ *
+ * @author ActiveEon Team
+ * @since 01/03/17
+ */
+@Component
+public class AzureScaleSetProvider extends AzureProvider {
+
+    /*
+     * Assumptions for the following code:
+     *
+     * 1 NodeSource (infra) <-> 1 scale set </-> 1 resource group
+     *
+     */
+
+    @Getter
+    protected final String type = "azureScaleSet";
+
+    private final Logger logger = Logger.getLogger(AzureScaleSetProvider.class);
+
+    @Override
+    public Set<Instance> createInstance(Infrastructure infrastructure, Instance instance) {
+
+        // Initialize Azure service connector
+        Azure azureService = azureServiceCache.getService(infrastructure);
+
+        // Retrieve general information, tags & options
+        String instanceTag = Optional.ofNullable(instance.getTag())
+                                     .orElseThrow(() -> new RuntimeException("ERROR missing instance tag/name from instance: '" +
+                                                                             instance + "'"));
+        List<Tag> tags = tagManager.retrieveAllTags(instance.getOptions());
+        Optional<Options> options = Optional.ofNullable(instance.getOptions());
+
+        // Initialize bunch of variables for Azure resources creation
+        final int azureResourcesUUID = Math.abs(infrastructure.hashCode()); // Make it distinguable and findable (OK since 1 NS <-> 1 Scale Set)
+        final String azureLoadBalancerName = "LB-" + azureResourcesUUID;
+        final String azureFrontendName = "FE-" + azureResourcesUUID;
+        final String azureBackendPoolName = "BE-SSH-" + azureResourcesUUID;
+        final String azureTCPProbeName = "PTCP-" + azureResourcesUUID; // useless but required
+        final String azureHTTPLoadBalancingRule = "RHTTP-" + azureResourcesUUID; // useless but required
+        final String azureNATPoolSSHName = "NATSSH-" + azureResourcesUUID;
+        final String azureNATPoolPNPName = "NATPNP-" + azureResourcesUUID;
+        final String azureNATPoolPNPSName = "NATPNPS-" + azureResourcesUUID;
+        final String azureVNetName = "VNET-" + azureResourcesUUID;
+        final String azureSubnetName = "Front-end";
+        final String azureVnetAddressSpace = "172.16.0.0/16";
+        final String azureVnetAddressPrefix = "172.16.1.0/24";
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this IP "172.16.1.0" address configurable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1313)